### ISSUE TITLE
Create Menu molecule & Navigation organism

### ISSUE THREAD

**Issue by alexgibson** [mozilla/protocol#182]
Creatting this issue to track work for the new navigation component.

Sketch file plus wireframes: https://share.goabstract.com/967ec565-e446-4b30-aa03-65b3dd169d36

Content & structure: https://docs.google.com/document/d/1NNqfr5vtKj1tDz7HSBPTlWI2HFcXks7twzfZYLoK-hg/edit?usp=sharing

**Comment by alexgibson** [comment]
Still very much a WIP:

- https://demo2--mozilla-protocol.netlify.com/demos/menu.html
- https://demo2--mozilla-protocol.netlify.com/patterns/molecules/menu.html

Note: so far this is just the Menu molecule, which will be one component of the wider Navigation organism (which will also feature the Mozilla logo & download button component).

**Comment by alexgibson** [comment]
Demo links updated to include navigation organism ^

**Comment by thejeffreytaylor** [comment]
Going to move this to blocked, as two incoming high-priority projects have taken the time I was going to give to terminology.

**Comment by alexgibson** [comment]
There's a second google doc with an outline for copy / links which seems more up to date than the one linked earlier: https://docs.google.com/document/d/1g0N8oy0WSOY218LJjDIslBbVOmNaBY4Agnfo-cvp6wA/edit

**Comment by alexgibson** [comment]
Things we still need to close this demo out:

[ ] - Final copy / links
[ ] - Mobile high-fidelity mockups
[ ] - Desktop design needs a close button for the menu panel

**Comment by michaelmccombie** [comment]
The repeated Firefox logo in the first design were starting to become an issue. So I drew some illustrations to replace those logos. I created a new branch with the illustrations and a new hover state for the menu links which is visually similar to the hover state for the card component. Could probably do with a code review and some refactoring. 

https://github.com/mozilla/protocol/pull/190

![image](https://user-images.githubusercontent.com/5924575/43383468-f3930390-938f-11e8-8e7b-3d84c5e6c711.png)

**Comment by michaelmccombie** [comment]
**Summary of Changes:**
- Illustrations for the six links in the Firefox tab
- The illustrations are 64px instead of 24px 
- The links have the gray box-shadow to mimic the card
- The copy has been added back in and the description font-size was changed in the card to march the other links.

![jrbgs8x2zq](https://user-images.githubusercontent.com/5924575/43384508-1a0f822a-9393-11e8-9bf1-96915c4c17d1.gif)

**Comment by alexgibson** [comment]
Hmm, not sure we can have links within links like this. These large icons are also going to take up a lot of real estate at smaller screen sizes, so we may have to either shrink or hide them at smaller screen sizes.

**Comment by alexgibson** [comment]
Oh, nevermind - the border hover isn't a linkt it's just a visual effect?

**Comment by alexgibson** [comment]
I'll be honest, I find the card-like hover state here a little confusing from a UI perspective. With the original card component it indicates a clickable link, but here that's not quite true. For menu items that have multiple links, the border hover state can be visible even when a link is not being hovered. I'm a little wary this is going to create mis-clicks.

**Comment by alexgibson** [comment]
> I'll be honest, I find the card-like hover state here a little confusing from a UI perspective. With the original card component it indicates a clickable link, but here that's not quite true. For menu items that have multiple links, the border hover state can be visible even when a link is not being hovered. I'm a little wary this is going to create mis-clicks (will still give it a shot to see how it feels).

Ok, I think I have something that works OK here. The key is making sure there is zero margin or negative space in/around the title link. 👍

**Comment by alexgibson** [comment]
Demo updated with the latest content / illustrations as per above:

https://demo2--mozilla-protocol.netlify.com/demos/navigation.html

Note: for smaller screen sizes, I decreased the icon size to 32px.

**Comment by michaelmccombie** [comment]
Thanks for working on this. You got the hover states correct, that's how I had them working on my branch. My only suggestion would be to add a 16px padding-bottom to `mzp-c-menu-item-list` to add some space between the list and the grey drop-shadow.

**Comment by michaelmccombie** [comment]
Sorry to drag this out, but there are a couple of changes to make to the demo. 
These changes were temporarily made on demo-2 but I reverted those commits so that branch should be back to normal.

**Copy**
- The "Projects" tab should be relabeled "Innovations"

**SCSS**
- Remove `border: 2px solid $color-gray-30;` on .mzp-c-menu-panel
- Replace the box-shadow on .mzp-c-menu-panel with `box-shadow: rgba(0, 0, 0, 0.2) 0 8px 32px;`

**Assets**
- Updated Firefox Tab Icons
- A new image for the "Pocket by Firefox" card
- A new image for the "Mozilla Web VR" card
- A new image for the "The Mozilla Festival" card

**Comment by alexgibson** [comment]
Thanks for this @michaelmccombie 👍 

A couple of things to discuss here:

1. Icon sizes are not quite uniform. Currently icons are squares of 64px x 64px, however these latest assets we seem to be using smaller icons (e.g. for Firefox), with arbitrary padding in each image to make them 64px squared. This is going to make images more fiddly and less flexible for use elsewhere if we start adding padding around things for very specific use cases. I think we need to pick an icon size, and make sure that icons are sized properly the neccessary width/height.

2. The new box shadow looks great, however removing the border makes the upward arrow pretty hard to see when the menu is over a white background. I think we may need a new arrow asset here with some faux-shadow baked in, as I can't apply CSS to a border here.

**Comment by alexgibson** [comment]
This is now updated on demo: https://demo2--mozilla-protocol.netlify.com/demos/navigation.html

I picked a new icon size of 40px x 40px, based on the amount of padding being added to the Firefox logo. Of course, this now means the illustration icons are smaller than they should be, because they also have some arbitrary padding around them.

**Comment by alexgibson** [comment]
The latest design & copy is now updated and on demo2: https://demo2--mozilla-protocol.netlify.com/demos/navigation.html

Some notes:

- I changed the demo page to use a light background, and added some lorem ipsum page text. I think this will help give us a better idea for how the navigation will look & feel on the majority of mozorg pages.
- The demo still needs updated icons, and a new card image for Firefox Add-ons.
- Once we finalize design, there are still some things I need to do to get this in a finished state:
  - [ ] Add final assets to the protocol-assets repository.
  - [ ] Tidy up CSS & HTML example code.
  - [ ] Finish documentation
  - [ ] Add RTL styles for the navigation.
  - [ ] Final round of cross-browser testing.

Once the above checklist is complete, I'll open a pull request to add this to Protocol.

**Comment by michaelmccombie** [comment]
Hey @alexgibson I submitted a PR on the assets repo with the new icons. https://github.com/mozilla/protocol-assets/pull/13

There are also a couple of minor CSS changes we were hoping to make.
1. Change the icon size to 24px by 24px
2. Change the border-top in `.mzp-c-menu-panel` from 2px to 1px
3. Change the box-shadow in `.mzp-c-menu-panel` from `-webkit-box-shadow: 0 6px 12px -6px rgba(0, 0, 0, 0.2);` to `box-shadow: 0 16px 16px -16px rgba(0, 0, 0, 0.3);`

**Comment by michaelmccombie** [comment]
Also here is an image for the Add-Ons card.
[add-ons-image.zip](https://github.com/mozilla/protocol/files/2296305/add-ons-image.zip)

**Comment by michaelmccombie** [comment]
Jeffrey noticed a minor typo. The word "Windows" has a capital "I".

![image](https://user-images.githubusercontent.com/5924575/44432857-a80e1380-a558-11e8-84a5-d83df6541799.png)
